File: 0004 two sum/main.cpp
//Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.
//You may assume that each input would have exactly one solution,
//and you may not use the same element twice. You can return the answer in any order.


#include <iostream>
#include <vector>
using namespace std;

int twosum(vector <int> num, int target){
    int k = 0;
    for (int i=0; i<num.size(); i++){
        for (int j=0; j<i; j++){
            if (num[i] + num[j] == target){
                    printf("Output:[%d, %d] \n", j, i);
            }
        }
    }
}

int main(){
    twosum({1,5,8,9,20,50}, 59);
}

#########################
# Output:[3, 5]         #
#########################

File: 0004 two sum/main.py
def number(array, target):
    for i in array:
        for j in array:
            if i + j == target:
                return [i, j]
            else:
                continue

def findplace(array, target_array, place):
    for i in range(len(array)):
        if array[i] == target_array[place]:
            return i
        else:
            continue

def twosum(array, target):
    try:
        target_array = number(array, target)
        first_element = findplace(array, target_array, 0)
        second_element = findplace(array, target_array, 1)
        print("Input: num =", array, ", target = ",target)
        print("Output: ",[first_element, second_element])
    except:
        print("sum error, place type correct sum number")

if __name__ == "__main__":
    twosum([1, 4, 7, 10, 11, 12], 17)

    
#####################################################
#Input: num = [1, 4, 7, 10, 11, 12] , target =  17  #
#Output:  [2, 3]                                    #
#####################################################
